### DISCUSSION TITLE
Game Design Question:  Do you support large resolutions or is a max resolution size OK?

### DISCUSSION THREAD

**Discussion by rdarnold** [MonoGame/MonoGame#8558] [General]
So for indie games, do you think it's ok to make a max resolution at, say (for example), 1280 x 720, or should all of our games allow a full-screen version no matter what the resolution?  I'm finding it difficult to get artwork at these high resolutions that are coming out in monitors these days.  I could just let people maximize and let the artwork scale beyond target resolution but I'm wondering if that's how others are handling it?

**Comment by KelsamGames** [comment]
I'd say this is mostly a matter of opinion, constraints, and target platforms.

The constraint in this case being that you're using pre-existing artwork: if you don't really have a way around that, then I'd say your best bet is to use what you have and find the best way to upscale as gracefully as possible when necessary. That can be done via RenderTargets/shaders/etc, or even just passing desirable enum values to SpriteBatch so it scales using X pre-packaged algorithm (i.e. when making a pixel art game, you at least want it to maintain hard-edged pixels even if you're not going so far as to ensure perfect pixel-multiple scaling and such).

To the "wondering how others are handling it" what I can say is this: we specifically write our games with what we consider to be a reasonable future-proofed-ish target resolution "max" and we generally design our UI/menus/etc to support a wide range of aspect ratios, so we never have to black bar our games when scaling. That isn't to say that is always the right way, just what we like best and generally target for our own games. That said, we create all our own artwork, usually in vector, and rasterize it for the game so it can handle up to 4k without having to stretch anything beyond its natural resolution (probably, at least in the near-term). THEN, there are considerations to be made there too. Because all our sprites/textures/etc support up to 4k, but we still want them looking their best at 1080p, we end up with pretty large compiled assets due to being huge to begin with, and the added load of mipmapping and such. For example: one of our 2D games like this that has quite a few sprites/textures/etc needs about 3GB storage and usually requires nearly 4GB of memory to run (it loads nearly all textures into memory at the beginning to avoid long load screens during play). Now, had we targeted 1080p max res for that game, we could have easily cut those numbers by two thirds at the very least, but because we knew our target machines weren't going to have an issue with that storage/memory usage, it wasn't really a problem that needed solving. Even when porting the game for mobile and platforms like Atari VCS, it just didn't come into play much.

All I mean to say with all that is, there are a number of things to consider when making your final choices about target resolution and handling various screen aspect ratios and whether or not your game plays well in a window. All of which should be weighed and thought about specifically in the context of the game you're making. If you write your game targetting 1280x720 and force it run in a window, will people that use a 4k monitor hate it? If so, how likely is that to happen in your target platform(s)? If you let it run in full screen on a 4k monitor and it scales up your sprites made for a 720p rez, do they look horrible? If so, maybe you could look into some super-sampling methods/shaders to address it. This is one of the big benefits of writing your games with MonoGame, is you don't have to adhere to a "one size fits all" solution, you can decide exactly how your game should handle this.

**Reply by rdarnold** [reply]
Cool, thank you so much, that's very helpful!  So basically you write for all resolutions.  That's been my feeling as well, but I have enough doubts that it helps a lot to have some verification that others are at least doing the same thing.
